src/main/java/com/example/spring/validator/UniqueIsbnValidator.java

```diff
@@ -0,0 +1,24 @@
+package com.example.spring.validator;
+
+import com.example.spring.repositories.book.BookRepository;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class UniqueIsbnValidator implements ConstraintValidator<UniqueIsbn, String> {
+
+    private final BookRepository bookRepository;
+
+    @Override
+    public void initialize(UniqueIsbn constraintAnnotation) {
+    }
+
+    @Override
+    public boolean isValid(String isbn, ConstraintValidatorContext context) {
+        return isbn != null && !bookRepository.existsByIsbn(isbn);
```

Comment:
if you would like to validate isbn, it's better to use regexp. 
You should already have unique constraint for isbn in model so it's redundant validation
```
@Column(unique = true, nullable = false)
private String isbn;
```